Src/Support/Google.Apis.Auth/OAuth2/ServiceAccountCredential.cs

```diff
@@ -344,7 +362,10 @@ private Task<string> GetOrCreateJwtAccessTokenAsync(string authUri)
                 }
                 // Create a new JWT.
```

Comment:
You need to update the cache code above this line.
If this credential has explicit scopes, then you don't need to cache the tokens per authUri because that's being ignored in the tokens. You also dont' need to cache them per scopes, because the credential is inmutable. Basically, if this credential has explicit scopes you can use the same token for all auth uris, you only need to cache one. I suggest that in that case you use a constant string as the key for the cache, something like "SCOPED_TOKEN".

Reply:
Good point, thanks. I was thinking about the cache but i was trying to understand if we need a composite key. And we don't. 

And you are right, if we keep it as is - it store token per each authUri, which is not needed.

Reply:
updated the logic and added test